src/function/debounce.ts

```diff
@@ -17,25 +18,57 @@
  *
  * // Will not log anything as the previous call is canceled
  * debouncedFunction.cancel();
+ *
+ * // With AbortSignal
+ * const controller = new AbortController();
+ * const signal = controller.signal;
+ * const debouncedWithSignal = debounce(() => {
+ *  console.log('Function executed');
+ * }, 1000, signal);
+ *
+ * // Will cancel the debounced function call
+ * controller.abort();
  */
-export function debounce<F extends (...args: any[]) => void>(func: F, debounceMs: number): F & { cancel: () => void } {
+export function debounce<F extends (...args: any[]) => void>(
+  func: F,
+  debounceMs: number,
+  abortSignal?: AbortSignal
```

Comment:
To receive signals, what about we follow the [fetch API's option](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#aborting_a_fetch)?

```tsx
interface DebounceOptions {
  signal?: AbortSignal;
}

export function debounce<F extends (...args: any[]) => void>(
  func: F,
  debounceMs: number,
  { signal }: DebounceOptions = {}
) { /* ... */ }
```

Reply:
It was reflected in [b6ed097](https://github.com/toss/es-toolkit/pull/45/commits/b6ed097b8c2d62b77b42e736ccb5a57d08f63c0a)